packages/commonwealth/vite.config.ts

```diff
@@ -0,0 +1,124 @@
+import react from '@vitejs/plugin-react-swc';
+import path from 'path';
+import { Alias, defineConfig, loadEnv } from 'vite';
+import { createHtmlPlugin } from 'vite-plugin-html';
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+const projectRootDir = path.resolve(__dirname);
+
+function createClientResolver(folder: string): Alias {
+  const find = new RegExp(`^${folder}/(.*)$`);
+  return {
+    find,
+    replacement: path.resolve(
+      projectRootDir,
+      'client',
+      'scripts',
+      folder,
+      '$1',
+    ),
+  };
+}
+
+// https://vitejs.dev/config/
+export default defineConfig(({ command, mode }) => {
+  const env = loadEnv(mode, process.cwd(), '');
+  return {
+    plugins: [
+      react(),
+      createHtmlPlugin({
+        entry: `./${command === 'serve' ? 'client/' : ''}scripts/index.tsx`,
+        template: './client/index.html',
+      }),
+      tsconfigPaths(),
+    ],
+    publicDir: 'static',
+    server: {
+      port: 8080,
+      host: 'localhost',
+      proxy: {
+        '/api': {
+          target: env.BACKEND_PROXY_URL || 'http://localhost:3000',
+          changeOrigin: true,
+          secure: false,
+        },
+      },
+    },
+    resolve: {
+      alias: [
+        {
+          // matches only non-relative paths that end with .scss
+          find: /^([^.].*)\.scss$/,
+          replacement: path.resolve(
+            projectRootDir,
+            'client',
+            'styles',
+            '$1.scss',
+          ),
+        },
+        createClientResolver('hooks'),
+        createClientResolver('navigation'),
+        createClientResolver('state'),
+        createClientResolver('views'),
+        createClientResolver('controllers'),
+        createClientResolver('models'),
+        createClientResolver('helpers'),
+      ],
+      extensions: ['.ts', '.tsx', '.js', '.jsx'],
+    },
+    define: {
+      global: {},
+      'process.env.NODE_ENV': JSON.stringify('development'),
+      'process.env.SERVER_URL': JSON.stringify(env.SERVER_URL),
```

Comment:
not sure if already done, but should we do env validation? What if the SERVER_URL isn't present?

Reply:
Sorry, don't know how this got in here. I will remove it.